File: Cpp/1790_Check_if_One_String_Swap_Can_Make_Strings_Equal.cpp
class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {
        if (s1 == s2) return true;
        
        vector<int> frequency1(26, 0);
        vector<int> frequency2(26, 0);
        int diff = 0;
        
        for(int i = 0; i < s1.size(); i++) {
            if (s1[i] != s2[i]) ++diff;
            ++frequency1[s1[i] - 'a'];
            ++frequency2[s2[i] - 'a'];
        }
        
        if(diff != 2) return false;
        
        for(int i = 0; i < 26; ++i) {
            if(frequency1[i] != frequency2[i]) return false;
        }
        
        return true;
    }
};

File: Kotlin/1790_Check_if_One_String_Swap_Can_Make_Strings_Equal.kt
class Solution {
    fun areAlmostEqual(s1: String, s2: String): Boolean {
        val diff = s1.indices.filter { s1[it] != s2[it] }
        
        if(diff.isEmpty()) return true
        if(diff.size == 2) {
            val (index1, index2) = diff
            return s1[index1] == s2[index2] && s1[index2] == s2[index1]
        }
        
        return false
    }
}